### ISSUE TITLE
ft_get_si behavior differs depending on inputs

### ISSUE THREAD

**Issue by sckott** [ropensci/fulltext#68]
character input - works fine

```r
ft_get_si("10.1371/journal.pone.0126524", "pone.0126524.g005.jpg", "epmc")
#> [1] "/var/folders/gs/4khph0xs0436gmd2gdnwsg080000gn/T//Rtmp8JHVqv/10.1371_journal.pone.0126524_pone.0126524.g005.jpg"
```

ft_get input - works with numbers, not IDs

```r
x <- ft_get('10.1371/journal.pone.0126524')
ft_get_si(x, "pone.0126524.g005.jpg", from = "epmc")
#> Error in func(x, si, save.name = save.name, cache = cache, vol = vol,  : 
#>  PLoS download requires numeric SI info In addition: Warning message:
#>In ft_get_si.ft_data(x, "pone.0126524.g005.jpg", from = "epmc") :
#>  Ignoring 'from' argument
```

Same thing for output from `ft_search()`. But they both work passing in just numeric number, as you say in the docs. I wonder if this is intended?

**Comment by sckott** [comment]
@willpearse

**Comment by willpearse** [comment]
I can't check right now, but I'm 99% sure it's related to how I index to get the right file from the vector.

For some of the plugins, I imagine I have been a bit... aggressive in my ```is.numeric``` / ```is.character``` -ing, and I'm happy to go back through and fix that up a touch!...

**Comment by sckott** [comment]
thanks!

**Comment by sckott** [comment]
@willpearse any thoughts on this?

**Comment by willpearse** [comment]
Right, this is as "intended", but perhaps confusing... ```ft_data``` and ```ft_search``` ignore (with a warning, as in your example) any ```from``` argument. Thus while the DOIs are the same in both examples, the actual calls are different.

I can't remember precisely why I did this, but it's defensively written and I know I did it on purpose. I sense it's because ```ft_data``` and ```ft_search``` returns information on where each thing the user has searched from has been found: I can't see a reason the user would want to try EPMC when they can get direct from the source by default.

The simplest fix would be to change the warnings to errors (https://github.com/ropensci/fulltext/blob/master/R/ft_get_si.R#L111 and https://github.com/ropensci/fulltext/blob/master/R/ft_get_si.R#L121) errors; that would have made this easier to see in this example, I think.

**Comment by sckott** [comment]
Okay. Maybe we can have `ft_get_si` fail better, I like it when we can fail well. 

Another question: In the man file for the `from` param you have 

> Only use this argument if \code{x} isn't a vector of DOI(s). 

can you make clear in what cases you should use the `from` param in `ft_get_si` ?

**Comment by willpearse** [comment]
Done and done https://github.com/willpearse/fulltext/commit/2f724e842761c6806e1254cf6a3de78c3ae62dbd

**Comment by sckott** [comment]
PR?

**Comment by willpearse** [comment]
Done! It's small; hopefully it's fine :D

**Comment by sckott** [comment]
closed from #77
